Adyen/UI/View Controllers/SearchViewController/SearchViewController.swift

```diff
@@ -0,0 +1,192 @@
+
+    internal lazy var loadingView: UIActivityIndicatorView = {
+        let loadingView = UIActivityIndicatorView(style: .whiteLarge)
+        loadingView.color = .Adyen.componentLoadingMessageColor
+        loadingView.hidesWhenStopped = true
+        return loadingView
+    }()
+    
+    internal lazy var searchBar: UISearchBar = {
+        let searchBar = UISearchBar()
+        searchBar.searchBarStyle = .prominent
+        searchBar.placeholder = searchBarPlaceholder
+        searchBar.isTranslucent = false
+        searchBar.backgroundImage = UIImage()
+        searchBar.barTintColor = style.backgroundColor
+        searchBar.delegate = self
+        searchBar.translatesAutoresizingMaskIntoConstraints = false
+        return searchBar
+    }()
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = style.backgroundColor
+
+        view.addSubview(loadingView)
+        view.addSubview(emptyView)
+        
+        resultsListViewController.willMove(toParent: self)
+        addChild(resultsListViewController)
+        view.addSubview(resultsListViewController.view)
+        resultsListViewController.didMove(toParent: self)
+        resultsListViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        
+        searchBar.setContentCompressionResistancePriority(.required, for: .vertical)
+        view.addSubview(searchBar)
+        
+        setupConstraints()
+        updateInterface()
+        
+        // Allow for immediate population
+        lookUpAddress(for: "")
+    }
+    
+    override public func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        if shouldFocusSearchBarOnAppearance {
+            searchBar.becomeFirstResponder()
+        }
+    }
+    
+    private func setupConstraints() {
+        
+        NSLayoutConstraint.activate([
+            searchBar.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 8),
+            searchBar.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -8),
+            searchBar.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor, constant: 0),
+
+            resultsListViewController.view.topAnchor.constraint(equalTo: searchBar.bottomAnchor, constant: 0),
+            resultsListViewController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: 0),
+            resultsListViewController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 0),
+            resultsListViewController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: 0),
+            
+            emptyView.leadingAnchor.constraint(equalTo: view.layoutMarginsGuide.leadingAnchor, constant: 0),
+            emptyView.trailingAnchor.constraint(equalTo: view.layoutMarginsGuide.trailingAnchor, constant: 0),
+            emptyView.centerYAnchor.constraint(equalTo: view.layoutMarginsGuide.centerYAnchor, constant: 0)
```

Comment:
Can we use `UIStackView`? 🤔

Reply:
Yeah I was looking into it but in this specific case it's easier to do it using autolayout because we're hiding specific views - which indicates to the stack view that it should "remove" the hidden items - which in turn causes other layout issues unfortunately when it comes to e.g. the loading indicator

Reply:
🥲